src/functions/getNextUploadBatch/framework/handler.ts

```diff
@@ -32,19 +32,22 @@ export async function handler(event: APIGatewayEvent, fnCtx: Context): Promise<R
     return createResponse(validationResult.error, HttpStatus.BAD_REQUEST);
   }
 
-  await bootstrapConfig();
-  const batchPromise = getNextUploadBatch(batchSize, interfaceType);
-  await batchPromise.then((response) => {
-    // compress response
+  try {
+    await bootstrapConfig();
+    const response = await getNextUploadBatch(batchSize, interfaceType);
     nextBatchData = [
       ...response.map(((row) => {
         return gzipSync(JSON.stringify(row.test_result)).toString('base64');
       })),
     ];
-  }).catch((err) => {
-    console.error(err);
-  });
-  return createResponse(nextBatchData, HttpStatus.CREATED);
+    return createResponse(nextBatchData, HttpStatus.CREATED);
+  } catch (err) {
+    error('Unable to retrive a batch of results - ', enrichError(err, interfaceType, batchSize));
+    return createResponse(
+      { message: `Error trying retrive a batch of ${ batchSize } results for ${ interfaceType }` },
```

Comment:
And again

Reply:
probably same here... let's adjust the wording a bit

Reply:
I'm going to leave this for now. And put some work to improve it in a second ticket. Splitting out the errors and testing will take a few hours which risks us getting this into UAT